Message encryption

jborean93 opened the pull request:
Minor changes around how the library encrypts and decrypts messages so it can handle larger messages. Has also more easily exposed the TLS wrapping and unwrapping methods.

jborean93 commented:
Coverage results on Appveyor

![image](https://user-images.githubusercontent.com/8462645/27938193-bf095b2c-62ff-11e7-8044-1c8607894dc6.png)

nitzmahone commented on requests_credssp/credssp.py:
```diff
@@ -309,15 +310,55 @@ def _send_encrypted_credentials(self, response, **kwargs):
+        encrypted_data = b''
+
+        while True:
+            try:
```
Should probably have a sanity counter w/ forced exit on both of these loops around some multiple of len(data)- if the TLS session is borked without throwing (seen it happen, sadly), we'd technically go forever on this.

nitzmahone commented on requests_credssp/credssp.py:
```diff
@@ -309,15 +310,55 @@ def _send_encrypted_credentials(self, response, **kwargs):
+        data = b''
+
+        while True:
+            try:
```
sanity counter as above

nitzmahone commented on README.md:
```diff
@@ -67,10 +83,43 @@ r = requests.get("https://server:5986/wsman", auth=credssp_auth)
--------
+#### Message Encryption
+
+You can use this library to encrypt and decrypt messages sent ot and from the
```
s/ot/to/

jborean93 commented:
@nitzmahone changes have been made, let me know if you think it could be done differently.

jborean93 commented:
Hey @nitzmahone I've updated the PR and added some logic to calculate the trailer length here https://github.com/jborean93/requests-credssp/blob/f6d435a500439009a1bf26fca1d050c429149e28/tests/functional/test_credssp_connection.py#L113-L156. Unfortunately I couldn't find an easy way to do any of this stuff and is reliant on parsing the cipher name returned by OpenSSL. There are also various scenarios at play here but I have tested the following so far

* AES-GCM cipher which has a fixed trailer length
* TLS 1.2 with AES-CBC on different MAC hashes
* TLS 1.0 with AES-CBC on different MAC hashes

I need to do some more testing around this but any ideas on how to make this less hacky I am all ears.
